pycbc/inference/burn_in.py

```diff
@@ -28,6 +28,59 @@
 
 import numpy
 
+def ks_test(sampler, fp):
+    """Burn in based on whether the p-value of the KS test between the samples
+    at the last iteration and the samples midway along the chain for each
+    parameter is > 0.1 and < 0.9
+
+    Parameters
+    ----------
+    sampler : pycbc.inference.sampler
+        Sampler to determine burn in for. May be either an instance of a
+        `inference.sampler`, or the class itself.
+    fp : InferenceFile
+        Open inference hdf file containing the samples to load for determing
+        burn in.
+
+    Returns
+    -------
+    burn_in_idx : array
+        Array of indices giving the burn-in index for each chain.
+    is_burned_in : array
+        Array of booleans indicating whether each chain is burned in.
+    """
+    from scipy.stats import ks_2samp
+    from scipy import interpolate
+    nwalkers = fp.nwalkers
+    niterations = fp.niterations
+    burn_in_idx = numpy.repeat((niterations-1), nwalkers).astype(int)
+    # The chains should not be burned in by default. So is_burned_in is assigned
+    # a False value for all the walkers by default.
+    is_burned_in = numpy.zeros(nwalkers, dtype=bool)
```

Comment:
Set `burn_in_idx` and `is_burned_in` below, once you've figured out if the chain is burned in or not.